File: 1_year/LGDP/L3/PY/sala/3.py
print('\nDescrição:\n1-Número que será analisado a sua tabuada\n2-Primeiro valor da sequência\n3-Último valor da sequência\n4-Passo\n')
while True:
    x=list(map(int,input(': ').split()))
    if len(x)<1:
        break
    mudar=False
    if x[1]>x[2]:
        x[1]-=x[2]
        x[2]+=x[1]
        x[1]=x[2]-x[1]
        mudar=True
    temp=x[1] if not mudar else x[2]
    print()
    for i in range(x[1],x[2]+1,x[3]):
        print('{}*{} = {}'.format(x[0],temp,x[0]*temp))
        temp+=x[3] if mudar==False else -x[3]
    print()
print('Até!\n')

File: 1_year/LGDP/L3/C/sala/3.c
#include<stdio.h>
#include<string.h>
#include<stdbool.h>
#include<stdlib.h>
#include"../funcoes.h"
//
int main(){
    int*v,**receba,m;
    char*inp,*split;
    bool ctrl;
    printf("\nDescricao:\n1-Valor n que sera mostrado a sua tabuada, situada na faixa entre y e z\n2-Valor y\n3-Valor z\n4-Passo\n\n");
    for(;;){
        inp=(char*)malloc(1000*sizeof(*inp));
        v=(int*)calloc(4,sizeof(*v));
        receba=(int**)malloc(2*sizeof(int*));
        printf("-> ");
        fgets(inp,100,stdin);
        for(int i=0;split=strtok_r(inp," ",&inp);++i){
            ctrl=true;
            i==0?m=atoi(split):i>0?v[i-1]=atoi(split):m;
            (int)split[0]>47&&(int)split[0]<58||split[0]=='-'?ctrl=false:ctrl;
            if(ctrl)
                break;
        }
        if(ctrl)
            break;
        printf("%i %i\n",v[0],v[1]);
        looop(v,receba,0);
        for(int i=0;i<(v[1]-v[0])/v[2]+1;++i){
            printf("%i*%i = %.0lf\n",m,receba[0][i],(double)m*receba[0][i]);
        }
        printf("\n");
        free(inp);
        free(v);
        free(receba);
    }
    printf("Bye, bye!\n\n");
    return 0;
}